src/modules/project/pages/projectView/views/projectMainBody/components/ContributeButton.tsx

```diff
@@ -24,14 +24,17 @@ export const ContributeButton = (props: ButtonProps) => {
   const isFundingDisabled = !isActive(project.status)
 
   const isInProjectPage = location.pathname.includes(PathName.project) && project?.rewards?.length > 0
-
-  const isProjectAcceptedInOpenCommunityVoteGrant = project.grantApplications.some(
+  const acceptedApplication = project.grantApplications.find(
     (application) =>
       application.status === GrantApplicantStatus.Accepted &&
       application.grant?.__typename === 'CommunityVoteGrant' &&
       application.grant.status === GrantStatusEnum.FundingOpen,
   )
 
+  const isProjectAcceptedInOpenCommunityVoteGrant = Boolean(acceptedApplication)
+  const grantName =
+    acceptedApplication?.grant?.__typename === 'CommunityVoteGrant' ? acceptedApplication.grant.title : ''
```

Comment:
Is the ternary operation needed here? You could always set the title even if it's not set no?

Reply:
Small side note: the grant `name` is different from the grant `title`, one is unique, and the other is not. I'd suggest renaming the variable to `grantTitle` to avoid that confusion.

Reply:
Yes it is needed because I am not fetching the title for the `BoardVoteGrant` this mandates a _typename check, otherwise it will complain. Yes good good point in the var naming

Reply:
You can fetch the `title` in the Board Grant query to simplify this code. In this case, it doesn't cost much to fetch an extra field.

Reply:
I am not fetching at all the Board vote grants in project since it does not matter for the logic relating to showing or not the login prompt on the project page so I think the cost here is a bit more that just fetching an extra field